aimmo-game/simulation/worker_manager.py

```diff
@@ -18,14 +18,12 @@ class _WorkerManagerData(object):
     """
     This class is thread safe
     """
-
     def __init__(self, game_state, user_codes):
         self._game_state = game_state
         self._user_codes = user_codes
         self._lock = Semaphore()
 
     def _remove_avatar(self, user_id):
-        assert self._lock.locked
         self._game_state.remove_avatar(user_id)
         del self._user_codes[user_id]
```

Comment:
Removing the lock means that a user can get added between these two lines leaving us in an inconsistent state: the user is in the game but we deleted their code.